icu4c/source/common/cjmlbe.cpp

```diff
@@ -0,0 +1,457 @@
+    int32_t idx = 0;
+    while (arr[idx] != -1) {
+        result->append(arr[idx++]);
+    }
+    return result.orphan();
+}
+
+static UnicodeString* cloneString(const UnicodeString &str, UErrorCode &status) {
+    LocalPointer<UnicodeString> result(str.clone(), status);
+    return result.orphan();
+}
+
+static void deleteElement(void *obj) {
+    delete static_cast<Element *>(obj);
+}
+
+static bool isValid(const UnicodeString &s) {
+    return s.length() != 1 || s[0] != INVALID;
+}
+
+
+int32_t CjMLBreakEngine::divideUpDictionaryRange(UText *inText, int32_t rangeStart,
+                                                 int32_t rangeEnd, UVector32 &foundBreaks,
+                                                 const UnicodeString &inString,
+                                                 const LocalPointer<UVector32> &inputMap,
+                                                 UErrorCode &status) const {
+
+    if (U_FAILURE(status)) {
+        return 0;
+    }
+    if (rangeStart >= rangeEnd) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
+        return 0;
+    }
+
+    UVector32 boundary(inString.countChar32() + 1, status);
+    if (U_FAILURE(status)) {
+        return 0;
+    }
+    int32_t numBreaks = 0;
+    UChar32 ch;
+    UnicodeString index;
+    UVector elementList(&deleteElement, nullptr ,6, status);
+    if (U_FAILURE(status)) {
+        return 0;
+    }
+
+    for (int32_t i = 1; i < inString.length(); i++) {
+        // Remove the first element and append a new element starting from index 4(2+2).
+        if (elementList.size() > 0) {
+            elementList.removeElementAt(0);
+            ch = (i + 2) < inString.length() ? inString.char32At(i + 2) : INVALID;
+            index = (ch != INVALID) ? getUnicodeBlockIndex(ch, status) : UnicodeString(INVALID);
+            elementList.adoptElement(new Element(ch, index), status);
+        }
+        // Initiate the elementList by adding six Element objects.
+        else {
+            initElementList(inString, elementList, status);
+        }
+        if (U_FAILURE(status)) {
+            return 0;
+        }
+
+        evaluateBreakpoint (
+            ((Element*)elementList.elementAt(0))->ch,
+            ((Element*)elementList.elementAt(1))->ch,
+            ((Element*)elementList.elementAt(2))->ch,
+            ((Element*)elementList.elementAt(3))->ch,
+            ((Element*)elementList.elementAt(4))->ch,
+            ((Element*)elementList.elementAt(5))->ch,
+            ((Element*)elementList.elementAt(0))->index,
+            ((Element*)elementList.elementAt(1))->index,
+            ((Element*)elementList.elementAt(2))->index,
+            ((Element*)elementList.elementAt(3))->index,
+            ((Element*)elementList.elementAt(4))->index,
+            ((Element*)elementList.elementAt(5))->index,
+            i, numBreaks, boundary, status);
+
+        if (U_FAILURE(status)) {
+            return 0;
+        }
+    }
+    elementList.removeAllElements();
```

Comment:
again I do not understand why you need to call elementList here? the destructor of UVector will remove all the elements for you, right? if you want to regain the memory free from that, what. you should do is the wrap this part of code in a
```
 {
     UVector elementList(....)
     ...
 } 
```
block and while the code left the block the destructor of the UVector will be called to free the memory.

Reply:
Done.